### PR TITLE
Defaults params not applied if ctx.params == null

### PR DESCRIPTION
**Pull request by thib3113**
### :dart: Relevant issues

https://discord.com/channels/585148559155003392/585149564135145493/955759619207225384
With io, I'm calling $node.actions .
And it crash with : 
Cannot read property 'onlyLocal' of null
coming from : https://github.com/moleculerjs/moleculer/blob/422876c8f0142caa313a792f8069ac7aac12fb47/src/registry/action-catalog.js#L131 

Debugging a little, I found this : 
const res = check(ctx.params != null ? ctx.params : {}, { meta: ctx }); => (this will define the default params of an action ) :  https://github.com/moleculerjs/moleculer/blob/422876c8f0142caa313a792f8069ac7aac12fb47/src/validators/base.js#L102 . 

The problem here ( I think ) . Is the ctx.params != null ? ctx.params : {} => yes, it will pass an object to the check function, but didn't retrieve the default params ? 

And so, If I correctly understand how it works, 
const params = ctx.params != null ? ctx.params : {}
const res = check(params, { meta: ctx });
ctx.params = params;

will resolve this error ? ( on my setup => yes ) 

### :gem: Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### :scroll: Example code
```js
try {
    const res = await broker.call('$node.actions', null);
    console.log(res);
} catch (e) {
    console.log(e);
}
``` 

## :vertical_traffic_light: How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Tested manually on my setup (with example code) + io request
Any ideas of tests ?

## :checkered_flag: Checklist:

- [ ] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] **I have added tests that prove my fix is effective or that my feature works**
- [ ] **New and existing unit tests pass locally with my changes**
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/validators/base.js [hunk 1]
```diff
@@ -76,7 +76,9 @@ class BaseValidator {
 				if (action[paramName] && typeof action[paramName] === "object") {
 					const check = self.compile(action[paramName]);
 					return function validateContextParams(ctx) {
-						const res = check(ctx.params != null ? ctx.params : {}, { meta: ctx });
+						const params = ctx.params != null ? ctx.params : {};
+						const res = check(params, { meta: ctx });
+						ctx.params = params;
```
[the remark this answers is not in the document]
Comment (thib3113) [reply src/validators/base.js:81]: sorry, I don't understand ? wich line ?

[the remark this answers is not in the document]
Comment (icebob) [reply src/validators/base.js:81]: Line 81

[the remark this answers is not in the document]
Comment (icebob) [reply src/validators/base.js:81]: Sry, I was confused, I thought this line is removed by the PR :)

[the remark this answers is not in the document]
Comment (thib3113) [reply src/validators/base.js:81]: @icebob in the two modifications, I just moove the `ctx.params != null ? ctx.params : {}` to a variables . ( so if the check function update the object, we set the params with this object ... before, the change to the empty object was lost )

File: src/validators/base.js [hunk 2]
```diff
@@ -99,8 +101,9 @@ class BaseValidator {
 				if (event[paramName] && typeof event[paramName] === "object") {
 					const check = self.compile(event[paramName]);
 					return function validateContextParams(ctx) {
-						const res = check(ctx.params != null ? ctx.params : {}, { meta: ctx });
-
+						const params = ctx.params != null ? ctx.params : {};
+						const res = check(params, { meta: ctx });
```
[the remark this answers is not in the document]
Comment (Nyamkhuub) [reply src/validators/base.js:105]: > if you do this, and ctx.params == null, the defaults parameters will not be applied . ( that's exactly the base problem )

Thank your for your reply.  I got it. ;)

**Comment by thib3113** [comment]
@icebob any news on this ?

**Comment by icebob** [comment]
I'm not sure it's the good solution, because it changes the logic. E.g. we can't pass `null` as a value of `ctx.params`.
I mean you call `broker.call("hello.world", null)` and in your service the `ctx.params` will be `{}` instead of `null`.

**Comment by thib3113** [comment]
> I'm not sure it's the good solution, because it changes the logic. E.g. we can't pass `null` as a value of `ctx.params`. I mean you call `broker.call("hello.world", null)` and in your service the `ctx.params` will be `{}` instead of `null`.

Hum, yes, I understand ... But, what about the default from validator ?

As far as I remember, the problem is what if I didn't pass paremeters (I  need to recheck this) ? 
( so `ctx.params === undefined ` / `ctx.params == null` )
We can do a check and fill with an object if it's `== null` but `!== undefined` ?

**Comment by thib3113** [comment]
( I need to reopen this, but not with the master branch ) .

**Pull request closed without merging**